src/index.ts

```diff
@@ -422,8 +422,20 @@ export default runExtension({
       handler:
         ({ proccessBlockText, variables }) =>
         (arg, format = "(({uid}))") => {
-          const queryUid = variables[arg] || arg;
-          const parentUid = extractRef(queryUid);
+          const queryRef = variables[arg] || arg;
+          const aliasOrPage = window.roamAlphaAPI.data.fast.q(
+            `[:find ?uid :where [?b :block/uid ?uid] [or-join [?b] 
+                 [and [?b :block/string ?s] [[clojure.string/includes? ?s "{{query block:${queryRef}}}"]] ]
+                 ${getQueryPages(extensionAPI).map(
+                   (p) =>
+                     `[and [?b :node/title ?t] [[re-pattern "^${p.replace(
+                       /\*/,
+                       queryRef
+                     )}$"] ?regex] [[re-find ?regex ?t]]]`
+                 )}
+            ]]`
+          )[0]?.toString();
```

Comment:
I was hoping for the alias to be editable in the UI but that may not be necessary. At the very least, it should be viewable in the UI somehow. If it's editable, we would have to build out:
```
- {{query block}}
    - alias
        - example
```

which might be more trouble than it's worth. Your solution handles the issue rather quickly. Thoughts?

In terms of retrieval, this is 🎯

Reply:
Reflecting on it more, I think I prefer the approach you took over the one I had in mind. Only addition I'd make is to make that alias viewable in the query block somehow

Reply:
I could see this being nice to unify where to find alias across `Query Blocks`, `Query Drawer`, and `Query Pages`!

Reply:
Agreed! Very nicely done.

Reply:
Let me know if you want that done in this PR or if we should ship what we have and look at unifying it later.

I've converted the PR from a draft either way.

Reply:
no need for it to be done now this looks good to ship